zipline/pipeline/loaders/buyback_auth.py

```diff
@@ -0,0 +1,114 @@
+"""
+Reference implementation for EarningsCalendar loaders.
+"""
+
+from ..data.buyback_auth import (
+    CashBuybackAuthorizations,
+    ShareBuybackAuthorizations
+)
+from events import EventsLoader
+from zipline.utils.memoize import lazyval
+
+
+BUYBACK_ANNOUNCEMENT_FIELD_NAME = 'buyback_dates'
+SHARE_COUNT_FIELD_NAME = 'share_counts'
+CASH_FIELD_NAME = 'cash'
+
+
+class CashBuybackAuthorizationsLoader(EventsLoader):
+    """
+    Reference loader for
+    :class:`zipline.pipeline.data.earnings.CashBuybackAuthorizations`.
+
+    events_by_sid: dict[sid -> pd.DataFrame(knowledge date,
+    event date, cash value)]
+
+    """
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=CashBuybackAuthorizations):
+        super(CashBuybackAuthorizationsLoader, self).__init__(
+            all_dates,
+            events_by_sid,
+            infer_timestamps=infer_timestamps,
+            dataset=dataset
+        )
+
+    def get_loader(self, column):
```

Comment:
Ah, never mind. I guess that's what the second part of your comment addresses. Thanks!